src/python/deepgnn/graph_engine/snark/converter/writers.py

```diff
@@ -57,20 +51,22 @@ def __init__(
         self.node_alias: typing.Union[NodeAliasWriter, _NoOpWriter] = (
             _NoOpWriter()
             if skip_node_sampler
-            else NodeAliasWriter(self.folder, self.suffix, self.node_type_num)
+            else NodeAliasWriter(self.folder, self.suffix)
         )
         self.edge_alias: typing.Union[EdgeAliasWriter, _NoOpWriter] = (
             _NoOpWriter()
             if skip_edge_sampler
-            else EdgeAliasWriter(self.folder, self.suffix, self.node_type_num)
+            else EdgeAliasWriter(self.folder, self.suffix)
         )
 
-        self.node_count = 0
-        self.edge_count = 0
-        self.node_weight = [0.0] * self.node_type_num
-        self.node_type_count = [0] * self.node_type_num
-        self.edge_weight = [0.0] * self.edge_type_num
-        self.edge_type_count = [0] * self.edge_type_num
+        self.node_count: int = 0
+        self.edge_count: int = 0
+        self.node_type_num: int = -1
```

Comment:
why the starting value is -1? I expect an empty graph to have 0 node|edge_types.